### ISSUE TITLE
Mixed content libraries don't work

### ISSUE THREAD

**Issue by mcarlton00** [jellyfin/jellycon#46]
**Describe the bug**

Title

**Additional context**

Reported from agcrouton on Matrix

**Comment by Miwer** [comment]
Almost a year since this issue was opened. Is there any ongoing work to fix this? I really would like to go from Jellyfin to Jellycon, since it is more lightweight and faster than Jellyfin, and as far as I know doesn't hijack the kodi database like the Jellyfin plugin. But since I have mixed content libraries, it kinda blocks this transition.

**Comment by mcarlton00** [comment]
In an indirect sort of way, there has been progress.  But directly, no.  The reason for the delay has multiple layers.

The biggest of these is time/interest.  I maintain this one pretty much alone, and I only have a limited amount of time to devote here.  Most of that time obviously goes towards things that impact me, and I don't like/use mixed content for a variety of reasons that aren't really worth going into here.

The other big one is that I'm not particularly fond of a lot of the network flow in the addon.  It's a bit convoluted and doesn't flow well, so I've been wanting to redo what's likely not an insignificant amount of code that would have to be refactored at a later date.

The other other big one is that mixed content fundamentally works different than other library types, which creates all kinds of fun.  Right now the libraries are populated based on views from the server.  The relevant code to show these in the addon menu is https://github.com/jellyfin/jellycon/blob/master/resources/lib/menu_functions.py#L954-L990.  The important part of this is that `collection_type = view.get('CollectionType', None)` line.  The reason this is important is because mixed content/other library types don't have a collection type, so adding support for them will probably require a different method of creating everything in this menu.  Assuming you do get that menu created so  the mixed library shows up, then the next functions also need to get modified because they also depend on `CollectionType` to retrieve the appropriate items https://github.com/jellyfin/jellycon/blob/master/resources/lib/menu_functions.py#L1032-L1054

TLDR: It's coming eventually, but it's a pretty niche thing and making it work is probably going to be less than fun

**Comment by chicungunya** [comment]
Hello,
I'm interested too. It will be on the next update? Thanks a lot!

**Comment by mcarlton00** [comment]
I don't know when it will be ready.  I'm working on redoing the network code now, but my other comments about the difficulties in implementing mixed libraries still exist.
